cfe_internal/enterprise/federation/federation.cf

```diff
@@ -88,9 +119,22 @@ bundle agent transport_user
       depth_search => default:recurse_with_base("inf"),
       file_select => default:all,
       perms => default:mog( "600", $(user), "root" );
+    "$(home)/.ssh/authorized_keys"
+      create => "true",
+      edit_template_string => "{{#-top-}}{{{.}}}$(const.n){{/-top-}}",
+      template_data => @(cfengine_enterprise_federation:config.pubkeys),
+      template_method => "inline_mustache";
+
   commands:
+    # Generate a modern ssh keypair (2019-04)
+    "/usr/bin/ssh-keygen"
+      handle => "ssh_keys_configured",
+      args => "-N '' -o -a 150 -t $(ssh_key_type) -f $(ssh_priv_key)",
```

Comment:
Don't we need to drop `-o`? or was that not problematic for one env?

Reply:
Wasn't a problem in my testing with ubuntu 16. But you are right, depending on version it may be a problem. Is -o required for the ssh_key_type you chose?

Reply:
```
     -o	     Causes ssh-keygen to save private keys using the new OpenSSH for-
	     mat rather	than the more compatible PEM format.  The new format
	     has increased resistance to brute-force password cracking but is
	     not supported by versions of OpenSSH prior	to 6.5.	 Ed25519 keys
	     always use	the new	private	key format.
```

So, no, sicne i am using ed25519  `-o` shouldn't be requred